packages/javascript-sdk/src/config/index.ts

```diff
@@ -60,6 +67,38 @@ abstract class Config {
     this.options = { ...setDefaults(options) };
   }
 
+  /**
+   * @method setAsync - Asynchronously calls the WellKnown endpoint to collect the APIs for OAuth
+   * @param {AsyncConfigOptions} options - config options with wellknown endpoint URL
+   * @returns {Promise<void>} - Returns a success or failure message object
+   */
+  public static async setAsync(options: AsyncConfigOptions): Promise<void> {
+    if (!options.serverConfig.wellknown) {
+      throw new Error(
+        'Missing well-known property. Use `Config.set` method if not using well-known endpoint.',
+      );
+    }
+    if (options.serverConfig.baseUrl) {
+      console.warn(
+        'The baseUrl property passed in will be ignored, and replaced with well-known origin.',
+      );
+    }
+
+    // Fetch wellknown endpoint for OAuth/OIDC API URLs
+    const json = await fetchWellKnown(options);
```

Comment:
Do we want to handle a failure of reaching the endpoint?

Reply:
I don't think so, we'd end up catching the error just to throw the error we caught, a bit redundant. By not catching it, we just allow the error to go straight to the app layer/developer.